tests/tests/tier0/bluechi-monitor-service/python/monitor.py

```diff
@@ -0,0 +1,122 @@
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+import re
+import signal
+import subprocess
+import threading
+import time
+import unittest
+
+from bluechi.api import Node
+
+node_name_foo = "node-foo"
+
+service_simple = "simple.service"
+
+log_file = "/tmp/bluechictl-output.txt"
+
+
+class FileFollower:
+    def __init__(self, file_desc):
+        self.pos = 0
+        self.file_desc = file_desc
+
+    def seek(self):
+        self.file_desc.seek(self.pos)
+
+    def new_lines(self):
+        self.seek()
+        return '\n' in self.file_desc.read()
+
+    def close(self):
+        if self.file_desc:
+            self.file_desc.close()
+
+    def __iter__(self):
+        while self.new_lines():
+            self.seek()
+            line = self.file_desc.read().split('\n')[0]
+            yield line
+
+            self.pos += len(line) + 1
+
+
+class TestMonitorSpecificNodeAndUnit(unittest.TestCase):
+
+    def setUp(self) -> None:
+        self.bluechictl_proc = None
+        self.state_pat = re.compile(r"\s*ActiveState: (?P<active_state>[\S]+)\s*")
+        self.bluechictl_out = None
+
+        self.inactive = False
+        self.active = True
+
+    def timeout_guard(self):
+        time.sleep(10)
+        print(f"Loop timeout - service '{service_simple}' on node '{node_name_foo}' "
+              f"was not successfully restarted on time")
+        self.bluechictl_proc.send_signal(signal.SIGINT)
+        self.bluechictl_proc.wait()
+
+    def process_events(self):
+        out_file = None
+        try:
+            out_file = open(log_file, mode='w')
+
+            self.bluechictl_proc = subprocess.Popen(
+                ["/usr/bin/bluechictl", "monitor", f"{node_name_foo}", f"{service_simple}"],
+                stdout=out_file,
+                bufsize=1)
+
+            self.bluechictl_out = FileFollower(open(log_file, mode='r'))
+            events_received = False
+            while self.bluechictl_proc.poll() is None and not events_received:
+                for line in self.bluechictl_out:
+                    print(f"Evaluating line '{line}'")
+                    match = self.state_pat.match(line)
+                    if not match:
+                        print(f"Ignoring line '{line}'")
+                        continue
+                    active_state = match.group("active_state")
+                    if active_state == "inactive" and self.active and not self.inactive:
+                        # Active -> Inactive
+                        print(f"Service '{service_simple}' on node '{node_name_foo}' was stopped successfully")
+                        self.active = False
+                        self.inactive = True
```

Comment:
Could we add a `break` here as well to abort early when we have a hit?

Reply:
Hmm, what would it help here? We need to check all messages until we get to inactive -> active transition (to successfully pass the test) and we don't know if this message is in this iteration of received lines or in the iteration which comes after timeout.

Reply:
But there can only be one match per `bluechictl_out`, right? It would also be just a tiny optimization, so lets keep it like it is.